src/components/Widgets/RelationControl.js

```diff
@@ -32,36 +26,54 @@ class RelationControl extends Component {
     super(props, ctx);
     this.controlID = uuid.v4();
     this.didInitialSearch = false;
+    // use local state to store the field's display value
+    this.state = { displayValue: '' };
   }
 
   componentDidMount() {
     const { value, field } = this.props;
     if (value) {
       const collection = field.get('collection');
-      const searchFields = field.get('searchFields').toJS();
+      // only perform the initial query on `valueField`
+      const searchFields = [ field.get('valueField') ];
+      // the goal of this query is to fetch the linked `collection` item
+      // to then show the value of it `displayValue` field to the user
       this.props.query(this.controlID, collection, searchFields, value);
     }
   }
 
   componentWillReceiveProps(nextProps) {
     if (this.didInitialSearch) return;
-    if (nextProps.queryHits !== this.props.queryHits && nextProps.queryHits.get && nextProps.queryHits.get(this.controlID)) {
+    if (
+      nextProps.queryHits !== this.props.queryHits &&
+      nextProps.queryHits.get && nextProps.queryHits.get(this.controlID)
+    ) {
       this.didInitialSearch = true;
-      const suggestion = nextProps.queryHits.get(this.controlID);
-      if (suggestion && suggestion.length === 1) {
-        const val = this.getSuggestionValue(suggestion[0]);
-        this.props.onChange(val, { [nextProps.field.get('collection')]: { [val]: suggestion[0].data } });
+      // store query results
+      const matches = nextProps.queryHits.get(this.controlID);
+      // if there is only one item found
```

Comment:
We haven't officially set a code style guide, but I do want to keep comments looking proper. Please update the comments to follow [Airbnb's comment guidelines](https://github.com/airbnb/javascript#comments), which I've also been loosely following (exception being that I use the multi-line style for all comments, but that isn't required).